shap/plots/dependence.py

```diff
@@ -138,6 +142,15 @@ def convert_name(ind):
     if categorical_interaction and clow != chigh:
         bounds = np.linspace(clow, chigh, chigh - clow + 2)
         color_norm = matplotlib.colors.BoundaryNorm(bounds, colors.red_blue.N)
+        
+    # optionally add noise if feature has less than 20 unique values
+    if (noise > 0) and (len(set(xv)) < 20):
+        if noise > 1: noise = 1
+        xvals = list(set(xv))
+        smallest_diff = np.min([xvals[i] - xvals[i-1] for i in range(1, len(xvals))])
```

Comment:
```suggestion
        smallest_diff = np.min(np.diff(np.sort(xvals)))
```
You could use `np.min(np.diff(xval))` here to avoid the comprehension. Also `set(xv)` may not preserve ordering, so you may want to call `sort` on the `xvals`. If `xvals` isn't ordered then you might end up with a negative value for `smallest_diff`.